flink-ml-parent/flink-ml-lib/src/main/java/org/apache/flink/ml/common/matrix/BLAS.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.ml.common.matrix;
```

Comment:
Yes for matrices and vectors, but I think BLAS  should store separately from elements which it matches.
I thought about the next structure of packages:
```
org.apache.flink.ml.common
    core
          BLAS.java
          vector
                   DenseVector.java SparseVrctor.java ...
          matrix
                   DenseMatrix.java ....
```
or
```
org.apache.flink.ml.common
      vector
               DenseVector.java  ...
       matrix
               DenseMatrix.java ....
       core
               BLAS.java

```
instead of `core` could use `linalgebra`. 
The second approach personally for me better because in this way we can encapsulate algebra processor by making separate package for inner kitchen.

Reply:
@ex00 I am not sure if it is necessary to make the packages so fine grained. It seems a little over packaging given that we do not have that many classes and probably won't have much more classes in the future. @xuyang1706 can probably comment here because I think they have already implemented a pretty decent ml library with these classes.

I would imagine that in the end we will have the following structure. Personally I fee it is clean enough to me. If `matrix` is the part that makes it less accurate, we could rename it to something like `linalg` which indicates `linear algebra`.

```
org.apache.flink.ml.
      common
             linalg
                    DenseMatix, DenseVector, SparseVector, BLAS, .... (
             SOME_OTHER_SHARED_PACKAGE
      logisticregression
      knn
      kmeans
      svm
      ...
```

Reply:
yes, you're right, the structure was suggested from me is overkill.
Renaming the `matrix` package is fine for me.

Reply:
I have renamed the package "matrix" to "linalg", which is short hand for "linear algebra".